src/main/java/org/apache/sling/scripting/core/impl/DefaultSlingScript.java

```diff
@@ -732,7 +732,14 @@ Bindings verifySlingBindings(final SlingBindings slingBindings) throws IOExcepti
 
             ProtectedBindings protectedBindings = new ProtectedBindings(bindings, protectedKeys);
             for (BindingsValuesProvider provider : bindingsValuesProviders) {
+                long start = System.nanoTime();
                 provider.addBindings(protectedBindings);
+                long stop = System.nanoTime();
+                if (stop-start > (1000*1000)) { // 1 ms
```

Comment:
it would be good to use a named constant for this value, also to avoid duplicating it in the ScriptContextProvider